connaisseur/validators/notrayv1/notary.py

```diff
@@ -49,6 +50,28 @@ def __init__(self, name: str, host: str, pub_root_keys: list, **kwargs):
         self.pub_root_keys = pub_root_keys
         self.is_acr = kwargs.get("is_acr", False)
         self.is_cosign = kwargs.get("is_cosign", False)
+        self.auth = kwargs.get("auth", {})
+        self.selfsigned_cert = self.__create_selfsigned_cert(
+            kwargs.get("selfsigned_cert")
+        )
+
+    def __create_selfsigned_cert(self, cert: str):
+        if not cert:
+            return None
+
+        cert_path = self.SELFSIGNED_PATH.format(self.name)
+
+        if not safe_path_func(os.path.exists, "/app/connaisseur/certs", cert_path):
+            safe_path_func(
+                os.makedirs,
+                "/app/connaisseur/certs",
+                os.path.dirname(cert_path),
+                exist_ok=True,
+            )
+            with safe_path_func(open, "/app/connaisseur/certs", cert_path, "w") as file:
+                file.write(cert)
```

Comment:
This functionality did not previously exist. Why do we need it now if we didn't need it previously?

Reply:
Previously the selfsigned certificates were extracted from the configuration during the helm chart creation. This has been changed, so that we don't really care how the configuration looks like and just put everything into a ConfigMap, except for some specifi fields which are put into a Secret. Then inside Connaisseur we merge these 2 together to get back the original configuration. 

Thing is, the python `requests` module expects the certificate to be inside a seperate `.crt` file and we can't pass it as a string. That's why the certificate is stored in a file inside the container.

Reply:
Understood. -- Why do we call the whole thing `selfsigned_cert`? `cert` would be easier. Also, we don't know whether it is self-signed.